snet_cli/mpe_treasurer_command.py

```diff
@@ -0,0 +1,160 @@
+from snet_cli.mpe_channel_command import MPEChannelCommand
+from snet_cli.utils import compile_proto
+from pathlib import Path
+import sys
+import os
+import grpc
+from eth_account.messages import defunct_hash_message
+from snet_cli.utils_proto import import_protobuf_from_dir
+from snet_cli.utils import int4bytes_big
+from snet_cli.utils_agi2cogs import cogs2stragi
+import web3
+
+
+# we inherit MPEChannelCommand because we need _get_channel_state_from_blockchain
+class MPETreasurerCommand(MPEChannelCommand):
+
+    def _sign_message_list_unclaimed(self, mpe_address, current_block):
+        message =  self.w3.soliditySha3(
+                        ["string",           "address",   "uint256"],
+                        ["__list_unclaimed", mpe_address, current_block])
+        return self.ident.sign_message_after_soliditySha3(message)
+    def _sign_message_list_in_progress(self, mpe_address, current_block):
+        message =  self.w3.soliditySha3(
+                        ["string",             "address",   "uint256"],
+                        ["__list_in_progress", mpe_address, current_block])
+        return self.ident.sign_message_after_soliditySha3(message)
+
+    def _sign_message_start_claim(self, mpe_address, channel_id, channel_nonce):
+        message =  self.w3.soliditySha3(
+                        ["string",           "address",   "uint256",  "uint256"],
+                        ["__start_claim", mpe_address,   channel_id,   channel_nonce])
+        return self.ident.sign_message_after_soliditySha3(message)
+
+    # import protobuf and return stub and request class
+    def _get_stub_and_request_classes(self, service_name):
+        # Compile protobuf if needed
+        codegen_dir = Path.home().joinpath(".snet", "mpe_client", "control_service")
+        proto_dir   = Path(__file__).absolute().parent.joinpath("resources", "proto")
+        if (not codegen_dir.joinpath("control_service_pb2.py").is_file()):
+            compile_proto(proto_dir, codegen_dir, proto_file = "control_service.proto")
+
+        stub_class, request_class, _ = import_protobuf_from_dir(codegen_dir, service_name)
+        return stub_class, request_class
+
+    def _decode_PaymentReply(self, p):
+        return {"channel_id" :  int4bytes_big(p.channel_id), "nonce" : int4bytes_big(p.channel_nonce), "amount" : int4bytes_big(p.signed_amount), "signature" : p.signature}
+
```

Comment:
There is common code across methods with slight differences. We can consider writing common methods at a later point.